File: model-client/src/main/java/de/q60/mps/shadowmodels/runtime/model/persistent/Transaction.java
package de.q60.mps.shadowmodels.runtime.model.persistent;

import de.q60.mps.shadowmodels.runtime.model.IConcept;
import de.q60.mps.shadowmodels.runtime.model.INodeReference;

import java.util.stream.LongStream;

public abstract class Transaction implements ITransaction {
  protected final IBranch branch;

  public Transaction(IBranch branch) {
    this.branch = branch;
  }

  @Override
  public IBranch getBranch() {
    return branch;
  }

  @Override
  public boolean containsNode(long nodeId) {
    return getTree().containsNode(nodeId);
  }

  @Override
  public IConcept getConcept(long nodeId) {
    return getTree().getConcept(nodeId);
  }

  @Override
  public long getParent(long nodeId) {
    return getTree().getParent(nodeId);
  }

  @Override
  public String getRole(long nodeId) {
    return getTree().getRole(nodeId);
  }

  @Override
  public String getProperty(long nodeId, String role) {
    return getTree().getProperty(nodeId, role);
  }

  @Override
  public INodeReference getReferenceTarget(long sourceId, String role) {
    return getTree().getReferenceTarget(sourceId, role);
  }

  @Override
  public LongStream getChildren(long parentId, String role) {
    return getTree().getChildren(parentId, role);
  }

  @Override
  public LongStream getAllChildren(long parentId) {
    return getTree().getAllChildren(parentId);
  }
}

File: model-client/src/main/java/org/modelix/model/api/Transaction.kt
package org.modelix.model.api

import java.util.stream.LongStream

abstract class Transaction(override val branch: IBranch?) : ITransaction {

    override fun containsNode(nodeId: Long): Boolean {
        return tree!!.containsNode(nodeId)
    }

    override fun getConcept(nodeId: Long): IConcept? {
        return tree!!.getConcept(nodeId)
    }

    override fun getParent(nodeId: Long): Long {
        return tree!!.getParent(nodeId)
    }

    override fun getRole(nodeId: Long): String? {
        return tree!!.getRole(nodeId)
    }

    override fun getProperty(nodeId: Long, role: String?): String? {
        return tree!!.getProperty(nodeId, role)
    }

    override fun getReferenceTarget(sourceId: Long, role: String?): INodeReference? {
        return tree!!.getReferenceTarget(sourceId, role)
    }

    override fun getChildren(parentId: Long, role: String?): LongStream? {
        return tree!!.getChildren(parentId, role)
    }

    override fun getAllChildren(parentId: Long): LongStream? {
        return tree!!.getAllChildren(parentId)
    }
}
